packages/ionic/src/commands/deploy/add.ts

```diff
@@ -0,0 +1,104 @@
+import { CommandInstanceInfo, CommandLineInputs, CommandLineOptions, CommandMetadata } from '../../definitions';
+import { FatalException } from '../../lib/errors';
+import { runCommand } from '../../lib/executor';
+
+import { DeployConfCommand } from './core';
+
+export class AddCommand extends DeployConfCommand {
+  async getMetadata(): Promise<CommandMetadata> {
+    const dashUrl = this.env.config.getDashUrl();
+
+    return {
+      name: 'add',
+      type: 'project',
+      summary: 'Adds the Deploy plugin (cordova-plugin-ionic) to the project',
+      description: `
+This command adds the Deploy plugin (cordova-plugin-ionic) for both Cordova and Capacitor projects.
+
+For a Condova project it just takes care of running the proper cordova cli command with the submitted parameters.
+
+For a Capacitor project it runs all the steps necessary to install the plugin, sync with the native projects and
+add the configuration to the proper iOS and Android configuration files.
+      `,
+      footnotes: [
+        {
+          id: 'dashboard',
+          url: dashUrl,
+        },
+      ],
+      exampleCommands: [
+        '',
+        '--app-id="abcd1234" --channel-name="Master" --update-method="background"',
+        '--max-store=2 --min-background-duration=30',
+        '--app-id="abcd1234" --channel-name="Master" --update-method="background" --max-store=2 --min-background-duration=30',
+      ],
+      options: this.commandOptions,
+    };
+  }
+
+  protected buildCordovaDeployOptions(options: CommandLineOptions): string[] {
+    const optionsToCordova = {
+      'app-id': 'APP_ID',
+      'channel-name': 'CHANNEL_NAME',
+      'update-method': 'UPDATE_METHOD',
+      'max-store': 'MAX_STORE',
+      'min-background-duration': 'MIN_BACKGROUND_DURATION',
+      'update-api': 'UPDATE_API',
+    };
+    const outputOptions = [];
+    for (const [optionKey, cordovaKey] of Object.entries(optionsToCordova)) {
+      if (options[optionKey]) {
+        outputOptions.push(`--variable`);
+        outputOptions.push(`${cordovaKey}=${options[optionKey]}`);
+      }
+    }
+    return outputOptions;
+  }
+
+  async preRun(inputs: CommandLineInputs, options: CommandLineOptions): Promise<void> {
+    // check if it is already installed
+    const alreadyAdded = await this.checkDeployInstalled();
```

Comment:
cc @dwieeb

Reply:
It's a centralized way to statefully represent what's integrated with your app. If you can add and remove something from your app, and it's not fully represented by some other system, e.g. `package.json` and installing Cordova plugins/platforms, then it's probably an integration.

If we think that the Deploy integration is fully represented by simply having the deploy plugin installed, then that's fine. If there's any other piece we need, for example bootstrapping something in `app.module.ts`, we may want to consider making it an integration.

Reply:
I think for now I'm of the opinion that it's represented by the package.json particularly because you can remove it using npm or cordova cli too with no opportunity for for the CLI to detect it and currently that's actually what the docs represent. That might change overtime I suppose but traditionally it has been a cordova plugin managed by cordova similar to how you manage other plugins and now a capacitor plugin. I think the command it really a utility only command...a helper for making the plugin installation and configuration easier by helping automate the official install/management tools (cordova CLI/ npm/config.xml/Info.plist).